### ISSUE TITLE
web2 documentation should recommend putChild instead of setattr

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1850]
|<img alt="marienz's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| marienz reported|
|-|-|
|Trac ID|trac#1850|
|Type|enhancement|
|Created|2006-06-23 15:12:01Z|

http://twistedmatrix.com/projects/web2/documentation/howto/object-traversal.html says you should use setattr if you have children with characters like "." or "-" that are not valid in python identifiers. It should recommend putChild instead.

<details><summary>Searchable metadata</summary>

```
trac-id__1850 1850
type__enhancement enhancement
reporter__marienz marienz
priority__highest highest
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web2 web2
keywords__ 
time__1151075521000000 1151075521000000
changetime__1158423385000000 1158423385000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid commented|
|-|-|

That example is wrong in other ways as well, possibly i lost some data while converting it to lore.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid commented|
|-|-|

Actually on a second look setattr is the right thing if you wish to add a child_ attribute to the class.  putChild will only work on an instance.

branch in [branches/web2-objtraversal-fix-1850](branches/web2-objtraversal-fix-1850)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|


Looks pretty good. Except for this bit:

```
class DotChildren(resource.Resource):
    return http.Response(200, stream='<html><head><script type="text/javascript" src="scripts.js" /></head></html>')
```

You can't use 'return' outside of a function.  Please fix, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid commented|
|-|-|

I knew that still didn't look right.  Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_7"></a>|@dreid set status to `closed`|
|-|-|

(In [18177]) Fix broken example in the web2 object-traversal document.

This makes the example 1) valid python, 2) a working resource.Resource.  This also specifies when it is appropriate to use setattr vs when you should use the resources .putChild method.

Author: dreid
Reviewer: jml
Fixes [#1850](https://github.com/twisted/twisted/issues/1850)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
